public/app/core/utils/shortLinks.ts

```diff
@@ -0,0 +1,36 @@
+import memoizeOne from 'memoize-one';
+import { getBackendSrv, config } from '@grafana/runtime';
+import { AppEvents } from '@grafana/data';
+import appEvents from 'app/core/app_events';
+import { copyStringToClipboard } from './explore';
+
+function buildHostUrl() {
+  return `${window.location.protocol}//${window.location.host}${config.appSubUrl}`;
```

Comment:
nit: Not sure this if it helps in this case, but I think there is an Url class that handles building urls.

Reply:
I wasn't able to find Url/Link class where I could move this/use it from.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
haha no worries! I thought that we have some class in Grafana that is handling this. But yeah, I looked into URL class and didn't find anything that would be helpful and solved any of our problems that we are solving.